File: VB.NET/FindFiles.vb
Imports System.Collections.Generic
Imports System.IO

Friend Class FindFiles

    Public Shared GRAPHIC_FILES As String() = New String() {"*.bmp", "*.jpg", "*.jpeg", "*.jpe", "*.jfif", "*.tif", _
     "*.tiff", "*.png", "*.gif", "*.wmf", "*.emf", "*.pdf"}



	Public Shared Function Find(path As String, masks As String()) As String()
		Dim result As New List(Of String)()
		For Each mask As String In masks
			Dim filenames As String() = Directory.GetFiles(path, mask)
			For Each filename As String In filenames
				Dim lowerFilename As String = filename.ToLower()
				If Not result.Contains(lowerFilename) Then
					result.Add(lowerFilename)
				End If
			Next
		Next
		result.Sort()
		Return result.ToArray()
	End Function

End Class

File: CSharp/FindFiles.cs
using System.Collections.Generic;
using System.IO;

namespace BarcodeDemo
{
    internal class FindFiles
    {

        public static string[] GRAPHIC_FILES = new string[] {
            "*.bmp", "*.jpg", "*.jpeg", "*.jpe", "*.jfif", "*.tif", "*.tiff",
            "*.png", "*.gif", "*.wmf", "*.emf", "*.pdf"
        };



        public static string[] Find(string path, string[] masks)
        {
            List<string> result = new List<string>();
            foreach (string mask in masks)
            {
                string[] filenames = Directory.GetFiles(path, mask);
                foreach (string filename in filenames)
                {
                    string lowerFilename = filename.ToLower();
                    if (!result.Contains(lowerFilename))
                        result.Add(lowerFilename);
                }
            }
            result.Sort();
            return result.ToArray();
        }

    }
}
